StrongInject.Generator/DisposalLowerer.cs

```diff
@@ -4,16 +4,17 @@
 
 namespace StrongInject.Generator
 {
-    internal class DisposalLowerer
+    internal sealed record DisposalLowerer
```

Comment:
I don't particularly like making this a record just so we get `with` for free. I feel like record tells me something about the nature of the type, and this type feels not very recordy. WDYT?

Reply:
So far I've been letting the practical benefits tell me what is recordy and what isn't, so my personal definition of "recordy" is still forming. But I agree. If we could do `with` without making it a record, we probably wouldn't want to make this a record.

I can replace with a `.WithDisposeAsynchronously(bool)` method?

Reply:
Updated.